### PR TITLE
(feature) 344 regional payscale support

### PR DESCRIPTION
**Pull request by despo**
- Adds LocalAuthority and related models
- Imports LAs and associated regional pay band area(s)
- Updates payscale validation for minimum salary to look at school's regional payscale and remove `MIN_ALLOWED_SALARY` constant
- Adds school association with regional pay band (as some LAs belong to multiple regional pay band areas)
- Updates JobSpecification controller to pass in school_id, which is now required to perform validations
- Update SalaryValidator to read and utilise school#minimum_payscale_salary for minimum check
- Update payscale factory to generate better data for salary values
- Improve & simplify tests for `UpdateSchoolsDataFromSource`
- Import and associate both existing and new regional pay scale data
- Add seed data for development and associate generated schools with regions + regional payscales
- Expose TF rake task to load local authority data in production

### CODE REVIEW

**Comment by despo** [comment]
[Regional payscale support - investigation and development](https://trello.com/c/cZtrRobM/344-regional-payscale-support-investigation-and-development)

**Comment by despo** [comment]
[Update school fields to match current fields exposed by GIAS](https://trello.com/c/3WEEcyAm/362-update-school-fields-to-match-current-fields-exposed-by-gias)

File: app/jobs/update_schools_data_from_source_job.rb [hunk 1]
```diff
@@ -10,42 +10,48 @@ def perform
...
-        school.northing = row['Northing']
+      school.easting = row['Easting']
+      school.northing = row['Northing']
 
-        school.url = valid_website(row['SchoolWebsite'])
+      school.url = valid_website(row['SchoolWebsite'])
+      school.phase = row['PhaseOfEducation (code)'].to_i
 
-        school.phase = row['PhaseOfEducation (code)'].to_i
+      school.status = row['EstablishmentStatus (name)']
```
Comment (tahb) [inline app/jobs/update_schools_data_from_source_job.rb]: This looks to be a fix we'd like to keep as it's a separate fix to the focus of this pull request.

File: app/jobs/update_schools_data_from_source_job.rb [hunk 2]
```diff
@@ -82,5 +88,7 @@ def csv_url
 
   def valid_website(url)
     Addressable::URI.heuristic_parse(url).to_s
+  rescue Addressable::URI::InvalidURIError
```
Comment (tahb) [inline app/jobs/update_schools_data_from_source_job.rb]: Another fix we could cherry-pick onto a new branch to keep along with it's test and get merged in asap.

**Comment by tahb** [comment]
As mentioned after this meeting https://trello.com/c/SX0BPMur the decision was made to change approach as storing the salary information on pay scales isn't providing value but does present significant complexity.

**Pull request closed without merging**